src/main/java/fi/aalto/cs/apluscourses/model/Course.java

```diff
@@ -231,21 +191,21 @@ private static String getCourseName(@NotNull JSONObject jsonObject, @NotNull Str
         String pluginName = requiredPluginsJson.getString(pluginId);
         requiredPlugins.put(pluginId, pluginName);
       } catch (JSONException ex) {
-        throw new MalformedCourseConfigurationFileException(path,
+        throw new MalformedCourseConfigurationFileException(source,
             "Expected id-name-pairs in requiredPlugins object", ex);
       }
     }
     return requiredPlugins;
   }
 
-
-  @NotNull
+  @Nullable
   @Override
-  public Module getModule(@NotNull String moduleName) throws NoSuchModuleException {
+  public Module getModuleOpt(@NotNull String moduleName) {
```

Comment:
what does `Opt` part stand for in `getModuleOpt(...)`: optional, options?

Reply:
could you please also JavaDoc this method :smirk_cat:

Reply:
It stands for optional. Originally I named the method `getModuleOptional`, but I thought that would cause the expectation of `Optional<Module>`. Would `getModuleOptional` be a better name?

Reply:
Good point about Javadoc!